tap_salesforce/__init__.py

```diff
@@ -353,7 +356,7 @@ def do_sync(sf, catalog, state):
         )
 
         catalog_metadata = metadata.to_map(catalog_entry["metadata"])
-        replication_key = catalog_metadata.get((), {}).get("replication-key")
+        replication_key = catalog_metadata.get((), {}).get("valid-replication-keys")[0]
```

Comment:
I love the commit message :)

Reply:
But I don't understand, are we making it DRY or is it just an irony? :D

Reply:
If think it wouldn't hurt us to create a function doing this `catalog_metadata.get((), {}).get("valid-replication-keys")[0]`

Reply:
she's making it DRY later (not in this commit) - right now she is just fixing it